### DISCUSSION TITLE
Room for whislists and Ideas is open

### DISCUSSION THREAD

**Discussion by mmattel** [mmattel/EET-Solmate#53] [Wishlist and Ideas]
Please add all your stuff here 😄 
This makes things easier to discuss and track.

**Comment by BetaTobi** [comment]
Would it be possible to add a timestamp on the terminal outputs?
![image](https://github.com/mmattel/EET-Solmate/assets/74064975/4ec80bf9-ca22-4df0-8b40-2aa9aa50c971)

This would help, to identify a potential issue.

**Reply by mmattel** [reply]
> Would it be possible to add a timestamp on the terminal outputs

I will consider this as it is for the terminal output a good tip. Maybe as an option controlled by a setting.
Additionally, I am thinking about making the whole stuff a HA addon by building a container. But this is some work as the HA documentation (in general) has really a lot of room for improvement. Writing the code is not that challenging than getting the knowledge what is needed to get it to work. Many trials and errors - therefore I do a lot of code documentation.

**Reply by mmattel** [reply]
@BetaTobi I have implemented this with v6 😄

**Reply by BetaTobi** [reply]
Top! I just read it :-)
Many thanks!

**Reply by BetaTobi** [reply]
![image](https://github.com/mmattel/EET-Solmate/assets/74064975/1dedac14-1c59-4e33-b5a5-a16d1df077e3)

it works :)

**Comment by BetaTobi** [comment]
And another question:
Would it be possible to add the version of THIS script (today: 5.0.0) to the MQTT entities?
This would also help to check if the update worked.

**Reply by mmattel** [reply]
> Would it be possible to add the version of THIS script

I like the idea, good point. I will add that.

**Reply by mmattel** [reply]
This will come with the next release, most likely a minor version.
There was too much work that needed to be released now. I have not forgotten this.

**Reply by mmattel** [reply]
Done, see: [v6.1.0](https://github.com/mmattel/EET-Solmate/releases/tag/v6.1.0)

**Reply by BetaTobi** [reply]
![image](https://github.com/mmattel/EET-Solmate/assets/74064975/6f124e79-882b-4d0a-9d15-fc69d8bc1e18)

Juhuu :-) 👍

**Comment by mmattel** [comment]
Here is an update on my attempt providing [esham](https://github.com/mmattel/EET-Solmate) as HA Add-On.
After many hours of pre-preperation, I summarize my findings:

* Beside that the HA documentation about [creating an addon](https://developers.home-assistant.io/docs/add-ons/tutorial) is very basic, I stumbled over a major issue.
* I, and many other HA users when reading the community, are using a [container installation](https://www.home-assistant.io/installation/).
As addons are also containers, these are (currently) excluded as option when you already have a dockerized setup...
* This leads to the point, that I would have to create a complete new and independent HA dev environment using own hardware or a dedicated non dockerized VM.

Because this is quite intense and time consuming, I have to postpone creating a HA addon.

The current two installation methods (independent, via HA shell access) are imho sufficient enough to satisfy using the software knowing that the shell access method is for enhanced users...

Here is a good summary of reasons not providing [addons in a HA container setup](https://community.home-assistant.io/t/addons-for-docker-installation/436190/4).

Maybe [HACS](https://developers.home-assistant.io/docs/creating_component_index) is a possibility but I have to check that when time permits...

**Reply by Fox7AT** [reply]
Servus,

first of all, thanks for your work.
I have only partial IT knowledge. I think many people would appreciate a easy setup possibility using HACS (I would too).

**Comment by mmattel** [comment]
Hello @Fox7AT 

I have long investigated using HACS but it turned out that only AppDaemon (AD) is doable. This has three fundamental causes: 1.) esham is self healing by its design. Under the hood, so many things can and do happen you would not believe. You get about 100% uptime if you do not fiddle around in HA or your OS/hardware which all will often cause the need for a reboot. 2.) you can have multiple independent instances of esham if you have more than one Solmate. 3.) There are many ways how HA can be installed and esham needs to be available for all of them. The [guide](https://github.com/mmattel/EET-Solmate?tab=readme-ov-file#installation-options) to install esham via AD is pretty straight forward as it uses things you get with HA and which is OS independent. I agree that using an easier method installing esham would be welcomed, but this is not possible because of the design and restrictions of HA.

Please let me know how it is going and if you need help to pass the steps described. If you are on Matrix, you can also reach me out there: `@mmattel:matrix.org`.
